network_simulator/packet.py

```diff
@@ -53,6 +53,7 @@ def __init__(self, packet_id=None, flow_id=None, source_id=None,
                         start_time_sec=start_time_sec,
                         size_bits=DATA_PACKET_SIZE_BITS)
 
+    # TODO(david): Need __eq__ here to do "packet in list()" in flow.py.
```

Comment:
You can implement this yourself. Just have a function in Packet that returns False with RoutingPackets (should never have to check these for equality afaik), and for other Packets checks type equality and ID equality.